invenio_collections/percolator.py

```diff
@@ -29,12 +31,16 @@ def new_collection_percolator(target):
     query = IQ(target.dbquery)
     for name in current_search.mappings.keys():
         if target.name and target.dbquery:
-            current_search.client.index(
-                index=name,
-                doc_type='.percolator',
-                id='collection-{}'.format(target.name),
-                body={'query': query.to_dict()}
-            )
+            try:
+                current_search.client.index(
+                    index=name,
+                    doc_type='.percolator',
+                    id='collection-{}'.format(target.name),
+                    body={'query': query.to_dict()}
+                )
+            except TransportError:
```

Comment:
After trying both approaches:

> (a) If you have less than 100+ collections then I would not use percolators.

Tried with `COLLECTIONS_USE_PERCOLATOR = False` but the queries then don't match as with Percolators. A query definition like `field.subfield:value` will match with percolators on but will not with `_find_matching_collections_internally`

> (b) Have you tried https://www.elastic.co/guide/en/elasticsearch/reference/current/search-percolate.html#_forcing_unmapped_fields_to_be_handled_as_strings ?

That works (hopefully without side effects). It means that, to start with, we need to add to [invenio-marc-21](https://github.com/inveniosoftware/invenio-marc21/tree/master/invenio_marc21/mappings/marc21) this option to the mappings so that installations that use that package don't fail when creating collections. Is that the preferred option?